repos/system_upgrade/el7toel8/libraries/rpms.py

```diff
@@ -16,3 +16,12 @@ def get_installed_rpms():
                                                                                               RC=err.exit_code)
         stdlib.api.current_logger().error(error)
         return []
+
+
+def create_lookup(model, field, key, context):
+    return {getattr(obj, key) for obj in getattr(next((m for m in context.consume(model)), model()), field)}
```

Comment:
Please, would it be possible to rewrite this oneliner into more simpler constructs? I get what you want to do here but it hurts my brain too much to unpack the logic behind this (or add docstring explaining what is going on in simple terms). Thanks

Reply:
Sure, sorry for not putting wip label, it was kind of not review ready